[KNI] pfpstatus: use record pkg

shajmakh opened the pull request:
Use the new package `record` to store the PFP status into a an object that would hold all recorded statuses.
This is preliminary step for recording dynamic number of statuses.

The AI-assisted code was only for unit testing the recorder updates.

Assisted-by: Cursor v1.2.2
AI-Attribution: AIA HAb Hin R Cursor v1.2.2 model:gpt-5 v1.0

shajmakh commented on pkg-kni/pfpstatus/pfpstatus_test.go:
```diff
@@ -35,43 +39,137 @@ func TestDumpNodeStatus(t *testing.T) {
 
-	// copypasted from example: https://pkg.go.dev/time#Parse - no spacial meaning
-	timeStamp, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
+	re, err := record.NewRecorder(10, 5, time.Now)
```
new recorder fails with one node, https://github.com/openshift-kni/debug-tools/blob/main/pkg/pfpstatus/record/recorder.go#L197

shajmakh replied:
I'm mistaken

shajmakh replied:
the failure happens for any number of nodes if the full capacity of the recorder is reached, fix in https://github.com/openshift-kni/debug-tools/pull/94

shajmakh commented:
/hold

shajmakh commented:
/unhold

ffromani commented on pkg-kni/pfpstatus/pfpstatus.go:
```diff
@@ -18,26 +18,23 @@ package pfpstatus
 	"github.com/go-logr/logr"
 
 	"github.com/k8stopologyawareschedwg/podfingerprint"
+	"github.com/openshift-kni/debug-tools/pkg/pfpstatus/record"
```
why don't we replace all this package with `debug-tools/pfpstatus`? the new package is an extension of this one `pkg-kni/pfpstatus`.

shajmakh replied:
I don't see direct `debug-tools/pfpstatus` https://github.com/openshift-kni/debug-tools, am I missing something?

ffromani replied:
...with `debug-tools/pkg/pfpstatus`... :)

shajmakh replied:
you're absolutely right, adjusted the code in the latest push. Thanks for pointing it out.

shajmakh committed:
[KNI] main: use `denug-tools/pfpstatus` to collect statuses

`debug-tools/pfpstatus` is an extension for `pkg-kni/pfpstatus` and
it manages the recording logic of a fixed amount of pfpstatuses and
manages dumping them into a dedicated file.
Import and use this new package and delete `pkg-kni/pfpstatus`.

Signed-off-by: Shereen Haj <shajmakh@redhat.com>

ffromani commented:
/approve
/lgtm

thanks!

ffromani commented:
/approve
